ecommerce/views.py

```diff
@@ -40,3 +39,114 @@ def update_profile(request):
         user_form = UserForm(instance=request.user)
         profile_form = ProfileForm(instance=request.user.profile)
     return render(request, 'ecommerce/profile.html', {'user_form': user_form,'profile_form': profile_form})
+
+@login_required
+@transaction.atomic
+def cart_add(request, pk):
+    if request.method == 'POST':
+        quantity = request.POST['quantity']
+    else:
+        quantity = 0
+
+    product_get = Product.objects.get(pk=pk)
+    if product_get:
+        product_stock = product_get.quantity
+    else:
+        product_stock = 0
+
+    in_cart = Cart.objects.filter(user=request.user, product__pk=pk)
```

Comment:
dùng first để lấy ra object luôn e nha
```suggestion
    in_cart = Cart.objects.filter(user=request.user, product__pk=pk).first()
```